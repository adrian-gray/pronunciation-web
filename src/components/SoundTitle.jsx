import React from 'react'

import { splitHilite } from './../utils/Utils'

export default function SoundTitle (props) {
  const { phoneme, str } = props

  return (
    <div className='title'>
      <h1>{splitHilite(phoneme)} - {splitHilite(str)}</h1>
    </div>
  )
}
