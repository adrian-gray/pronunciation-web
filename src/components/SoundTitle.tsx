import React from "react";

import SplitHilite from "./../components/SplitHilite";

export default (props: { phoneme: string; str: string }) => {
  const { phoneme, str } = props;

  return (
    <h2>
      <SplitHilite str={phoneme} /> - <SplitHilite str={str} />
    </h2>
  );
};
