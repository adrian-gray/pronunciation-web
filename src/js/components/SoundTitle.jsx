import React from 'react'

import { splitHilite } from './../Utils'

export default function SoundTitle (props) {
  const { phoneme, str } = props

  return <h1>{splitHilite(phoneme)} - {splitHilite(str)}</h1>
}
