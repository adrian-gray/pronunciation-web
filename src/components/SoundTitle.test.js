import React from "react";
import { render, cleanup } from "@testing-library/react";

import SoundTitle from "./SoundTitle";

afterEach(cleanup);

test("SoundTitle", async () => {
  const props = {
    phoneme: "Bob",
    str: "The cat in the hat"
  };

  const { container } = render(<SoundTitle {...props} />);

  expect({ container }).toMatchSnapshot();
});
