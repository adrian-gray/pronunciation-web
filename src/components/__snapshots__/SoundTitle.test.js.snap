// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SoundTitle 1`] = `
Object {
  "container": <div>
    <h3>
      <span>
        Bob
      </span>
       
       - 
      <span>
        The cat in the hat
      </span>
       
    </h3>
  </div>,
}
`;
